packages/block-editor/src/components/inserter/index.native.js

```diff
@@ -32,27 +33,51 @@ import { store as blockEditorStore } from '../../store';
 
 const VOICE_OVER_ANNOUNCEMENT_DELAY = 1000;
 
-const defaultRenderToggle = ( { onToggle, disabled, style, onLongPress } ) => (
-	<ToolbarButton
-		title={ _x( 'Add block', 'Generic label for block inserter button' ) }
-		icon={
-			<Icon
-				icon={ plusCircleFilled }
-				style={ style }
-				color={ style.color }
-			/>
-		}
-		onClick={ onToggle }
-		extraProps={ {
-			hint: __( 'Double tap to add a block' ),
-			// testID is present to disambiguate this element for native UI tests. It's not
-			// usually required for components. See: https://git.io/JeQ7G.
-			testID: 'add-block-button',
-			onLongPress,
-		} }
-		isDisabled={ disabled }
-	/>
-);
+const defaultRenderToggle = ( {
+	onToggle,
+	disabled,
+	style,
+	containerStyle,
+	onLongPress,
+	isDefaultView,
+} ) => {
+	// The "default view" refers to the editor's appearance when no blocks
+	// are currently selected. The "add block" button has a separate style
+	// for the "default view", which are added via the below "defaultViewProps"
+	// and "defaultViewText" variables.
+	const defaultViewProps = isDefaultView && {
+		icon: <Icon icon={ plus } style={ style } />,
+		customContainerStyles: containerStyle,
+		fixedRatio: false,
+	};
+	const defaultViewText = (
+		<Text style={ styles[ 'inserter-menu__add-block-button-text' ] }>
+			{ __( 'Add Blocks' ) }
+		</Text>
+	);
+
+	return (
+		<ToolbarButton
+			title={ _x(
+				'Add block',
+				'Generic label for block inserter button'
+			) }
+			icon={ <Icon icon={ plusCircleFilled } style={ style } /> }
+			onClick={ onToggle }
+			extraProps={ {
+				hint: __( 'Double tap to add a block' ),
+				// testID is present to disambiguate this element for native UI tests. It's not
+				// usually required for components. See: https://git.io/JeQ7G.
+				testID: 'add-block-button',
+				onLongPress,
+			} }
+			isDisabled={ disabled }
+			{ ...defaultViewProps }
+		>
+			{ isDefaultView && defaultViewText }
```

Comment:
👋 @SiobhyB , not sure if you have tried this already for fixing the button flashing issue, but wanted to share some of my findings: a bit of promising success by using the `shouldReplaceBlock` state variable in the Inserter class. Specifically, by adding a `&& ! shouldReplaceBlock` conditional. This way, the text label is not shown when the editor is in the process of replacing the default richtext block.

But as I said, that's half-success, since the toolbar styling is not correct, since that's governed by the `HeaderToolbar` component. Not sure yet how to pass such transitional state to the HeaderToolbar yet.

Hope this all helps!

Reply:
@hypest, thank you so much for this! 🙇‍♀️ 

I think you're right that the trickiest part to this is with updating the parent component. With that in mind, I've made a change to set `isDefaultView` as an editor setting, which I envisage as being similar to other editor settings that influence the editor's appearance, such as `fixedToolbar`, `reducedUI`, `focusMode`, etc. 

I've been able to get some code that's _functionally_ working with that approach, but it still has some issues. The biggest of which is: As my latest code changes involve an import of `@wordpress/editor` inside a `@wordpress/block-editor` file, the automated tests fail with the `Cannot read property 'SETTINGS_DEFAULTS' of undefined, js engine: hermes` error (same issue that ended up blocking this PR: https://github.com/WordPress/gutenberg/pull/35021).

I feel like this is on the right track, though, and it's a case of finding a better way to control the new setting's data. I'll keep looking into this, but want to be transparent that I don't think it's something that will be quick for me to get to the bottom of.

Reply:
> With that in mind, I've made a change to set isDefaultView as an editor setting, which I envisage as being similar to other editor settings that influence the editor's appearance, such as fixedToolbar, reducedUI, focusMode, etc

Hmm, feels to me that the this showing/hiding of the textual part of the inserter button is not a setting per se. I mean, it's not configuration that we are expecting to hold through the lifecycle of the editor instance. Instead, it's state that changes depending on the user/caret focus. Are there other similar examples already in place perhaps, where we use the settings for state like this?

Perhaps what we need to "copy" is the way the settings are implemented (hooks/events/etc) but not piggyback on the settings component itself.

That said, I wonder if @jhnstn you have thoughts on this?

Reply:
> Instead, it's state that changes depending on the user/caret focus. Are there other similar examples already in place perhaps, where we use the settings for state like this?

The only other one that comes to mid is that we use the settings to update `impressions`, which is then used to update parts of the native UI. It was seeing [this line of code](https://github.com/WordPress/gutenberg/blob/95cda71f51ccdb2d764d2c142e746517bdd5a2b6/packages/block-editor/src/components/inserter/index.native.js#L210) in the Inserter component that got me thinking of using the approach I've taken in the latest commits. It feels kind of similar to me in that the `impressions` are more of a state than a setting.

> Perhaps what we need to "copy" is the way the settings are implemented (hooks/events/etc) but not piggyback on the settings component itself.

I wonder if there might be a way to use `context` to change this data. 🤔  Looking forward to hearing any ideas from Jason!

Reply:
This issue was addressed in https://github.com/WordPress/gutenberg/pull/39726/commits/bc9b4ea72674bbe43d5070deb0239830897374c9.